core/dbt/parser/schemas.py

```diff
@@ -447,9 +464,55 @@ def _parse_generic_test(
             column_name=column_name,
             test_metadata=metadata,
         )
-        self.render_update(node, config)
+        # This is what causes all of the TestConfig from_dicts
+        # lots of time spent here.
+        # this is ConfiguredParser.render_update
+        # This goes through jinja rendering but throws away the
+        # rendered template. The point here is the side effects of
+        # the 'render_update' which update various macro-related
+        # things in the nodes. Have not identified where that
+        # actually comes from yet. Will flatten out all the
+        # methods to see if I can figure it out.
+        self.render_test_update(node, config, builder)
+
         return node
 
+    def render_test_update(self, node, config, builder):
```

Comment:
👁️ 👁️ 
.    👄    wow!

I just want to come back to this part of the code to better understand it. Flagging to revisit!